app/forms/work_package_types/subject_configuration_form.rb

```diff
@@ -44,7 +44,7 @@ class SubjectConfigurationForm < ApplicationForm
           checked: !subject_configuration_manual?,
           label: I18n.t("types.edit.subject_configuration.automatically_generated_subjects.label"),
           caption: I18n.t("types.edit.subject_configuration.automatically_generated_subjects.caption"),
-          disabled: !enterprise? && subject_configuration_manual?,
+          disabled: !enterprise?,
```

Comment:
:red_circle:  I think this behavioural change also needs to be reflected in our backend code: app/services/update_type_service.rb

There we are checking so far, that you can't change from manual to automatic without enterprise. Now that we disable changing the pattern without enterprise, we also need to prevent that on the backend side, don't we?

Reply:
Nice. I like the new implementation on the contract.

However, as I wrote above there is existing code to check this, but it's not in the contract but in the `UpdateTypeService` in a method called `validate_enterprise_action`.

Now that you added the new check to the type contract, I think the check in the UpdateTypeService can be removed.

Reply:
To be honest, I am confused where this leaves us.

There **is** a check for enterprisiness in that service. I'd be fine to delay removal of that code to "later" if you think that this is not in scope of this PR. But changing the business rules on when a change is enterprisy or not enterprisy is in scope of this PR, so if you decide to not remove the code over there, then shouldn't it be fixed at least?

Right now we have:
* A UI enforcing new enterprisiness rules (changed behaviour in this PR)
* A contract enforcing new enterprisiness rules (newly added in this PR)
* A service enforcing _old_ enterprisiness rules (not touched at all in this PR)